### PR TITLE
added redis

### PR DESCRIPTION
**Pull request by MRJCrunch**

### CODE REVIEW

File: generators/app/templates/app/libs/utils/redis.includeRedis-addon.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+/**
+ * Configured redis instance.
+ */
+
+var config = require('utils/config').get('redis'),
+    redis = require('redis'),
+    BPromise = require('bluebird');
```
Comment (fxlrnrpt) [inline generators/app/templates/app/libs/utils/redis.includeRedis-addon.js]: We have an abstraction wrapping promises. Use it

File: generators/app/templates/app/libs/initializers/redis.includeRedis-addon.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var logger = require('utils/logger').app,
+    config = require('utils/config'),
+    redis = require('utils/redis');
+
+var attempts = 50,
```
Comment (fxlrnrpt) [inline generators/app/templates/app/libs/initializers/redis.includeRedis-addon.js]: Move it to config

File: generators/app/templates/app/libs/initializers/redis.includeRedis-addon.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                redisClient.quit();
+                resolve();
+            })
+            .on('error', (err) => {
+                logger.error('Redis connecting -> error', err);
+                att++;
+                if(att >= attempts){
+                    logger.error('Redis connecting -> failed after %s attempts', attempts);
+                    reject(err);
+                    process.exit(1);
```
Comment (fxlrnrpt) [inline generators/app/templates/app/libs/initializers/redis.includeRedis-addon.js]: I'm not sure exiting right away is the best option

**Review by fxlrnrpt — CHANGES_REQUESTED** [review]

File: generators/app/templates/app/libs/initializers/redis.includeRedis-addon.js (hunk 2 above)
Comment (MRJCrunch) [reply generators/app/templates/app/libs/initializers/redis.includeRedis-addon.js]: Done

File: generators/app/templates/app/libs/utils/redis.includeRedis-addon.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    logger = require('utils/logger').utils,
+    redisConf = require('utils/config').get('redis'),
+    errors = require('errors'),
+    _ = require('lodash'),
+    redis = require('redis');
+
+promise.promisifyAll(redis.RedisClient.prototype);
+
+async function createClient() {
+    return await new Promise(function(resolve) {
```
Comment (fxlrnrpt) [inline generators/app/templates/app/libs/utils/redis.includeRedis-addon.js]: await new promise.Promise

**Review by fxlrnrpt — CHANGES_REQUESTED** [review]

File: generators/app/templates/app/libs/utils/redis.includeRedis-addon.js (hunk 4 above)
Comment (MRJCrunch) [reply generators/app/templates/app/libs/utils/redis.includeRedis-addon.js]: done

**Review by fxlrnrpt — APPROVED** [review]

**Review by fxlrnrpt — APPROVED** [review]

**Review by fxlrnrpt — APPROVED** [review]

**Pull request merged**